verl/utils/checkpoint/fsdp_checkpoint_manager.py

```diff
@@ -230,18 +259,34 @@ def save_checkpoint(self, local_path: str, hdfs_path: str = None, global_step: i
         with warnings.catch_warnings():
             warnings.simplefilter("ignore")
             with get_fsdp_state_ctx(self.model, StateDictType.SHARDED_STATE_DICT, state_dict_cfg, optim_cfg):
-                model_path = os.path.join(local_path, f"model_world_size_{self.world_size}_rank_{self.rank}.pt")
-                optim_path = os.path.join(local_path, f"optim_world_size_{self.world_size}_rank_{self.rank}.pt")
-                extra_path = os.path.join(local_path, f"extra_state_world_size_{self.world_size}_rank_{self.rank}.pt")
+                model_state_dict = self.model.state_dict()
+                optimizer_state_dict = self.optimizer.state_dict() if self.optimizer is not None else None
+                lr_scheduler_state_dict = self.lr_scheduler.state_dict() if self.lr_scheduler is not None else None
+
+                extra_state_dict = {
+                    "lr_scheduler": lr_scheduler_state_dict,
+                    "rng": self.get_rng_state(),
+                }
```

Comment:
These are duplicate with code below which checks `self.should_save_xxx` to decide whether to save this part.